test/e2e/util/util.go

```diff
@@ -146,132 +143,6 @@ func wrapErrorWithClusterIssuerStatusCondition(client clientset.ClusterIssuerInt
-	var certificate *v1.Certificate = nil
-	pollErr := wait.PollImmediate(500*time.Millisecond, timeout,
-		func() (bool, error) {
-			log.Logf("Waiting for Certificate %v condition %v=%v", name, condition.Type, condition.Status)
-			certificate, err := client.Get(context.TODO(), name, metav1.GetOptions{})
-			if nil != err {
-				return false, fmt.Errorf("error getting Certificate %v: %v", name, err)
-			}
-			if !apiutil.CertificateHasConditionWithObservedGeneration(certificate, condition) {
-				log.Logf("Expected Certificate %v condition %v=%v (generation >= %v) but it has: %v", name, condition.Type, condition.Status, condition.ObservedGeneration, certificate.Status.Conditions)
-				return false, nil
-			}
-			return true, nil
-		},
-	)
-	return certificate, wrapErrorWithCertificateStatusCondition(client, pollErr, name, condition.Type)
-}
-
-// WaitForCertificateEvent waits for an event on the named Certificate to contain
-// an event reason matches the supplied one.
-func WaitForCertificateEvent(client kubernetes.Interface, cert *v1.Certificate, reason string, timeout time.Duration) error {
-	return wait.PollImmediate(500*time.Millisecond, timeout,
-		func() (bool, error) {
-			log.Logf("Waiting for Certificate event %v reason %#v", cert.Name, reason)
-			evts, err := client.CoreV1().Events(cert.Namespace).Search(intscheme.Scheme, cert)
-			if err != nil {
-				return false, fmt.Errorf("error getting Certificate %v: %v", cert.Name, err)
-			}
-
-			return hasEvent(evts, reason), nil
-		},
-	)
-}
-
-func hasEvent(events *corev1.EventList, reason string) bool {
-	for _, evt := range events.Items {
-		if evt.Reason == reason {
-			return true
-		}
-	}
-	return false
-}
-
-// try to retrieve last condition to help diagnose tests.
-func wrapErrorWithCertificateStatusCondition(client clientset.CertificateInterface, pollErr error, name string, conditionType v1.CertificateConditionType) error {
-	if pollErr == nil {
-		return nil
-	}
-
-	certificate, err := client.Get(context.TODO(), name, metav1.GetOptions{})
-	if err != nil {
-		return pollErr
-	}
-
-	for _, cond := range certificate.Status.Conditions {
-		if cond.Type == conditionType {
-			return fmt.Errorf("%s: Last Status: '%s' Reason: '%s', Message: '%s'", pollErr.Error(), cond.Status, cond.Reason, cond.Message)
-		}
-	}
-
-	return pollErr
-}
-
-// WaitForCertificateToExist waits for the named certificate to exist
-func WaitForCertificateToExist(client clientset.CertificateInterface, name string, timeout time.Duration) error {
-	return wait.PollImmediate(500*time.Millisecond, timeout,
-		func() (bool, error) {
-			log.Logf("Waiting for Certificate %v to exist", name)
-			_, err := client.Get(context.TODO(), name, metav1.GetOptions{})
-			if errors.IsNotFound(err) {
-				return false, nil
-			}
-			if err != nil {
-				return false, fmt.Errorf("error getting Certificate %v: %v", name, err)
-			}
-
-			return true, nil
-		},
-	)
-}
-
```

Comment:
Great. I think it's better to delete these unused functions rather than let them rot while unused.